samples/application_development/external_lib/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+if(DEFINED ENV{BOARD})
+  set(BOARD $ENV{BOARD})
+else()
+  set(BOARD qemu_x86)
+endif()
+
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+project(NONE)
+
+target_sources(app PRIVATE src/main.c)
+
+# The external static library that we are linking with does not know
+# how to build for this platform so we export all the flags used in
+# this zephyr build to the external build system.
+#
+# Other external build systems may be self-contained enough that they
+# do not need any build information from zephyr. Or they may be
+# incompatible with certain zephyr options and need them to be
+# filtered out.
+zephyr_get_include_directories(includes)
+zephyr_get_system_include_directories(system_includes)
+zephyr_get_compile_definitions(definitions)
+zephyr_get_compile_options(options)
+
+string(REPLACE ";"  " "   includes ${includes})
+string(REPLACE "-I" " -I" includes ${includes})
+
+string(REPLACE ";"  " "               system_includes ${system_includes})
+string(REPLACE "-isystem" " -isystem" system_includes ${system_includes})
+
+string(REPLACE ";"  " "   definitions ${definitions})
+string(REPLACE "-D" " -D" definitions ${definitions})
+
+set(options_string "")
+foreach(opt ${options})
+  if(opt MATCHES "<COMPILE_LANGUAGE:")
+    # TODO: Support COMPILE_LANGUAGE generator expressions
+    continue()
+  endif()
+  set(options_string "${options_string} ${opt}")
+endforeach()
```

Comment:
can we put all of the above in some common place? I see the code is duplicated in the other sample, so it has the potential of  being simplified

Reply:
Yes, will do